File: shops/angular/src/app/shared/ui/product-card/product-card.ts
import { Component, Input } from '@angular/core';

/**
 * Compact catalog tile. Markup/styles: `templates/default/product_card`
 * (shared with the Leptos shop). Controller stays in this host.
 */
@Component({
  selector: 'rs-product-card',
  templateUrl: '../../../../../../../templates/default/product_card/product_card.html',
  styleUrl: './product-card.scss',
})
export class ProductCard {
  @Input({ required: true }) name!: string;
  @Input({ required: true }) slug!: string;
  @Input() description: string | null | undefined = null;
  @Input({ required: true }) detailHref!: string;
}

File: shops/leptos-rangular/src/components/product_card/product_card.rs
use leptos::prelude::*;
use rangular_aot::HostCell;
use rangular_host::{Host, HostError, Value};

include!(concat!(env!("OUT_DIR"), "/rangular/product_card_view.rs"));

/// One catalog tile from the shared theme (`templates/<theme>/product_card`).
#[component]
pub fn ProductCardPanel(
    name: String,
    slug: String,
    description: Option<String>,
    detail_href: String,
) -> impl IntoView {
    product_card_view(HostCell::new(ProductCardHost {
        name,
        slug,
        description,
        detail_href,
    }))
}

struct ProductCardHost {
    name: String,
    slug: String,
    description: Option<String>,
    detail_href: String,
}

impl Host for ProductCardHost {
    fn get(&self, name: &str) -> Option<Value> {
        match name {
            "name" => Some(Value::Str(self.name.clone())),
            "slug" => Some(Value::Str(self.slug.clone())),
            "description" => Some(Value::Str(self.description.clone().unwrap_or_default())),
            "detailHref" => Some(Value::Str(self.detail_href.clone())),
            _ => None,
        }
    }

    fn call(&mut self, _: &str, _: &[Value]) -> Result<Value, HostError> {
        Ok(Value::Unit)
    }
}
